client/src/main/java/io/cloudsoft/winrm4j/client/WinRmClient.java

```diff
@@ -737,6 +749,9 @@ public DeleteResponse call() {
                 assertFaultCode(soapFault, WSMAN_FAULT_CODE_SHELL_WAS_NOT_FOUND);
             }
         }
+        if (bus != null) {
```

Comment:
The default behaviour of CXF is to share the bus - that's correct. The problem is that the bus is shared with all users of CXF, not just the winrm4j client. You could end up sharing the bus with any services that are running in the application winrm4j is part of.
The changes here update winrm4j to use its own bus, per `WinRmClient`. This has two positive sides:
  * not sharing the bus with other users of CXF
  * we know when to shut down the bus - we are not leaving just threads running when using shared bus, it's keeping connections open which causes https://github.com/cloudsoft/winrm4j/issues/34.

I don't have performance numbers from the change, but I'd expect any changes here are dwarfed by the network IO that follows, moreover it's done once on creating a client.